java/ql/src/semmle/code/java/frameworks/spring/SpringWebUtil.qll

```diff
@@ -0,0 +1,241 @@
+        "org.springframework.web.util;UriBuilder;true;replaceQueryParams;;;Element of MapValue of Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;scheme;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;scheme;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;scheme;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilder;true;userInfo;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;userInfo;;;Argument[0];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilder;true;userInfo;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;UriBuilderFactory;true;builder;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;UriBuilderFactory;true;uriString;;;Argument[-1..0];ReturnValue;taint",
+        "org.springframework.web.util;UriComponents$UriTemplateVariables;true;getValue;;;MapValue of Argument[-1];ReturnValue;value",
+        "org.springframework.web.util;UriTemplateHandler;true;expand;;;Argument[-1..0];ReturnValue;taint",
+        "org.springframework.web.util;UriTemplateHandler;true;expand;(String,Map);;MapValue of Argument[1];ReturnValue;taint",
+        "org.springframework.web.util;UriTemplateHandler;true;expand;(String,Object[]);;ArrayElement of Argument[1];ReturnValue;taint",
+        "org.springframework.web.util;AbstractUriTemplateHandler;true;getBaseUrl;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;AbstractUriTemplateHandler;true;setBaseUrl;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;AbstractUriTemplateHandler;true;setDefaultUriVariables;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;ContentCachingRequestWrapper;false;ContentCachingRequestWrapper;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;ContentCachingRequestWrapper;false;getContentAsByteArray;;;Argument[-1];ReturnValue;taint",
+        // note for review: the `content` of this wrapper is determined by what's written to it; there's a function `getWriter()`
+        //   that gets the writer used to write to it; should I link these up outside of the csv models?
+        "org.springframework.web.util;ContentCachingResponseWrapper;false;ContentCachingResponseWrapper;;;Argument[0];Argument[-1];taint",
+        "org.springframework.web.util;ContentCachingResponseWrapper;false;getContentAsByteArray;;;Argument[-1];ReturnValue;taint",
+        "org.springframework.web.util;ContentCachingResponseWrapper;false;getContentInputStream;;;Argument[-1];ReturnValue;taint",
+        // note for review: adds a cookie (argument 1) to an `HttpServletResponse` (argument 0); whether a cookie should taint a response
+        //   is somewhat unclear to me.
```

Comment:
@atorralba maybe you can answer this?

Reply:
The only issue I can think of regarding arbitrary Cookies in responses is [CRLF Injection](https://owasp.org/www-community/vulnerabilities/CRLF_Injection)/[HTTP Response Splitting](https://owasp.org/www-community/attacks/HTTP_Response_Splitting)/[Header Injection](https://portswigger.net/kb/issues/00200200_http-response-header-injection), but we have [already discussed this](https://github.com/github/codeql/pull/5937) and the conclusion was that most frameworks already sanitize this internally, so there's no issue in adding unsanitized cookies to a response object.

Again, this could be relevant if the response object is used as a container (`addCookie(tainted)` -> `tainted = getCookie()` -> `sink(tainted)`, but I don't think this is going to be very common.

Reply:
I'm going to remove the entire cookie generator from these models; I also doubt that that will ever be used to propagate taint in any other way.